File: y2024_C#/AdventOfCode/General/Day.cs
using System.Runtime.CompilerServices;

namespace AdventofCode.General;

public abstract class Day {
    private const string InputDirectory = "/home/korbinian/code/Python/advent-of-code/y2024_C#/AdventOfCode/Inputs";
    protected List<string> Input;
    protected int ResultPartOne;
    protected int ResultPartTwo;

    protected Day(string inputFileName) {
        var inputPath = Path.Combine(InputDirectory, inputFileName);
        var inputArray = File.ReadAllLines(inputPath);
        Input = inputArray.ToList();
    }


    public abstract int PartOne();
    public abstract int PartTwo();
}

File: y2023_Java/src/_general/Day.java
package _general;

import java.util.Objects;

public abstract class Day {
    private static final String INPUT_FILENAME = "input.txt";
    protected String[] input;
    protected int resultPartOne;
    protected int resultPartTwo;

    public Day() {
        String inputPath = getInputPath();
        input = new FileInput(inputPath).getFileInput();
    }

    protected String getInputPath() {
        return Objects.requireNonNull(getClass().getResource(INPUT_FILENAME)).getPath();
    }
    public abstract int partOne();
    public abstract int partTwo();
}
